intro.md

```diff
@@ -1,9 +1,19 @@
 # Introduction
 
 ## Design patterns
-
-What are design patterns? What are idioms? Anti-patterns.
+When developing programs, we have to solve many problems. A program can be viewed as a solution to a problem. It can also be viewed as a collection of solutions to many different problems. All of these solutions work together to solve a bigger problem.
 
 ## Design patterns in Rust
 
-Why Rust is a bit special - functional elements, type system - borrow checker
+There are many problems that share the same form. While the details are different, since they have the same form they can be solved using the same methods.
```

Comment:
I think the reason behind the line 

```
Why Rust is a bit special - functional elements, type system - borrow checker
```

is that we should say that rust is not object oriented, so design patterns are different with respect to other object oriented languages.
No need to address this in this PR, but why might want to write this as a TODO.

Reply:
added it as a TODO in intro